validation.go

```diff
@@ -15,6 +16,16 @@ const (
 	defaultValidateThrottle    = 8192
 )
 
+// ValidationError is an error that may be signalled from message publication when the message
+// fails validation
+type ValidationError struct {
+	Reason string
```

Comment:
Nit: the nice way to do this is to make this "Reason" an error as well and expose it via `Cause()`. That way, one can check `xerrors.Is(error, ValidationError{})` and `xerrors.Is(error, UltimateCause{})`.

Reply:
In principle this is indeed quite nice, but it's quite likely unnecessary -- the Reason strings are exported and part of the public API for tracers.

We _could_ add named errors for all of them but this seems more work than what is worth right now.

If you feel strongly about it, let's make an issue and address it.